src/Search/SearchServiceProvider.php

```diff
@@ -40,41 +39,47 @@ public function register()
 
     public function registerUserGambits()
     {
-        $this->app->when(UserSearcher::class)
-            ->needs(GambitManager::class)
-            ->give(function (Container $app) {
-                $gambits = new GambitManager($app);
+        $gambits = AbstractSearcher::gambitManager(UserSearcher::class);
+        $gambits->setFullTextGambit($this->app->make(UserFulltextGambit::class));
+        $gambits->add($this->app->make(EmailGambit::class));
+        $gambits->add($this->app->make(GroupGambit::class));
```

Comment:
we'd be making the gambits in the registration phase, don't we usually avoid resolving classes too early ?

also why not inject the GambitManager like before ?

Reply:
Good point, this should be moved to boot. I'm pretty sure this means that the event BC layer isn't working either.

Reply:
I wanted to avoid contextual binding (when needs give), because if extensions want to add their own searchers, they'll also need to include a full service provider.

Although actually now that I think about it, every gambit registered via the `SimpleFlarumSearch` extender is associated with a Searcher class, so we could dynamically construct all the contextual bindings. That's actually quite a bit nicer, and cuts down a little bit on the evil static stuff.